habitat/tasks/nav/nav.py

```diff
@@ -576,18 +589,71 @@ def draw_source_and_target(self, episode):
             s_y - point_padding : s_y + point_padding + 1,
         ] = maps.MAP_SOURCE_POINT_INDICATOR
 
-        # mark target point
-        t_x, t_y = maps.to_grid(
-            episode.goals[0].position[0],
-            episode.goals[0].position[2],
-            self._coordinate_min,
-            self._coordinate_max,
-            self._map_resolution,
-        )
-        self._top_down_map[
-            t_x - point_padding : t_x + point_padding + 1,
-            t_y - point_padding : t_y + point_padding + 1,
-        ] = maps.MAP_TARGET_POINT_INDICATOR
+        for goal in episode.goals:
+            if goal.view_points is not None:
+                for view_point in goal.view_points:
+                    # mark view point
+                    t_x, t_y = maps.to_grid(
+                        view_point.agent_state.position[0],
+                        view_point.agent_state.position[2],
+                        self._coordinate_min,
+                        self._coordinate_max,
+                        self._map_resolution,
+                    )
+
+                    self._top_down_map[
+                        t_x - point_padding : t_x + point_padding + 1,
+                        t_y - point_padding : t_y + point_padding + 1,
+                    ] = maps.MAP_VIEW_POINT_INDICATOR
+
+        for goal in episode.goals:
+            # mark target point
+            t_x, t_y = maps.to_grid(
+                goal.position[0],
+                goal.position[2],
+                self._coordinate_min,
+                self._coordinate_max,
+                self._map_resolution,
+            )
+
+            self._top_down_map[
+                t_x - point_padding : t_x + point_padding + 1,
+                t_y - point_padding : t_y + point_padding + 1,
+            ] = maps.MAP_TARGET_POINT_INDICATOR
+
+            sem_scene = self._sim.semantic_annotations()
+            object_id = goal.object_id
+            assert int(sem_scene.objects[object_id].id.split("_")[-1]) == int(
+                goal.object_id
+            )
+
+            center = sem_scene.objects[object_id].aabb.center
+            import itertools
+
+            x_len, _, z_len = sem_scene.objects[object_id].aabb.sizes / 2.0
+            corners = [
+                center + np.array([x, 0, z])
+                for x, z in itertools.product([-x_len, x_len], [z_len, -z_len])
```

Comment:
Perhaps `for x, z in [(-x_len, -z_len), (-x_len, z_len), (x_len, z_len), (x_len, -z_len), (-x_len, -z_len)]` would be better?  That would get rid of `corners[2], corners[3] = corners[3], corners[2]` and `corners.append(corners[0])` (took me a bit to figure out what those where doing).